test/unit/frontend/github-setup.test.ts

```diff
@@ -22,27 +22,53 @@ describe("Github Setup", () => {
 	});
 
 	describe("#GET", () => {
-		it("should return HTML when jiraHost is missing", async () =>
+		beforeEach(async () => {
+			await Installation.create({
+				jiraHost,
+				clientKey: "abc123",
+				secrets: "def234",
+				sharedSecret: "ghi345",
+			});
+		});
+
+		it("should return redirect to github oauth flow for GET request if token is missing", async () =>
 			supertest(frontendApp)
 				.get("/github/setup")
-				.expect(res => {
-					expect(res.status).toBe(200);
-					expect(res.headers["content-type"]).toContain("text/html");
+				.set(
+					"Cookie",
+					getSignedCookieHeader({
+						jiraHost,
+					})
+				)
+				.expect((res) => {
+					expect(res.status).toBe(302);
```

Comment:
I am curious how does redirect happen? I see that /github/setup endpoint code always returns the handlebars template

(Does it happen somewhere in interceptors?)

Reply:
It's related to the middleware we are now passing to github/setup and the GH token check.